React/Views/RCTShadowView.m

```diff
@@ -86,11 +86,46 @@ static void RCTProcessMetaProps(const float metaProps[META_PROP_COUNT], float st
   : 0;
 }
 
+/*
+ * Sets the measure function on the cssNode, when the
+ * aspectRatio rectangle is set. This helps calculate
+ * the height of the node based on width of parent and
+ * maintains the aspect ratio of the node.
+ */
+static css_dim_t RCTAspectRatioMeasure(void *context, float width)
+{
+  RCTShadowView *shadowView = (__bridge RCTShadowView *)context;
+  if (isnan(width)) {
+    width = shadowView.aspectRatio.width;
+  }
+  
+  CGFloat computedHeight = 0.0f;
+  if (shadowView.aspectRatio.width > 0) {
+    computedHeight = width * (shadowView.aspectRatio.height / shadowView.aspectRatio.width);
+  }
+  
+  css_dim_t result;
+  result.dimensions[CSS_WIDTH] = width;
+  result.dimensions[CSS_HEIGHT] = computedHeight;
+  return result;
+}
```

Comment:
Ah, good catch. Unfortunately this needs work on the `css-layout` side. Currently because this was intended for text layout, it only deals with height based on width scenario...